contracts/Gateway.sol

```diff
@@ -101,43 +107,47 @@ contract Gateway is ProtocolVersioned, Owned {
         workers = _workers;
         bounty = _bounty;
         uuid = _uuid;
-
+        conversionRate = _conversionRate;
+        conversionRateDecimals = _conversionRateDecimals;
     }
 
     /**
      *  @notice External function requestStake.
      *
-     *  @dev In order to request stake the staker needs to approve Gateway contract for stake amount.
-     *       Staked amount is transferred from staker address to Gateway contract.
+     *  @dev In order to request stake the staker needs to approve Gateway contract for request stake amount.
+     *       Maximum possible staked amount is transferred from staker address to Gateway contract given the conversion rate.
```

Comment:
Please change to
> Given the conversion rate and the requested stake amount as an upper limit, the maximum stakeable amount is calculated transferred.
It calculates the stakeable amount so that loss of funds is minimised.

Or who has a better idea? Cc @pgev @jasonklein